Packs/Slack/Integrations/SlackEventCollector/SlackEventCollector.py

```diff
@@ -0,0 +1,142 @@
+        """
+        Drops from the API response of the current fetch the events that were already fetched in the previous run.
+        Args:
+            events (list): The raw events from the API.
+            last_run (dict): If exists, contains the `oldest` and `id` values of the most recent event fetched
+               in the previous run.
+
+        Returns:
+            (list) All the events that occurred *after* the record stored in the lastRun object.
+        """
+        if last_run and events:
+            if events[0].get('date_create') == last_run.get('oldest'):
+                for idx, event in enumerate(events):
+                    if event.get('id') == last_run.get('last_id'):
+                        return events[idx + 1:]
+        return events
+
+
+def test_module_command(client: Client) -> str:
+    client.test()
+    return 'ok'
+
+
+def get_events_command(client: Client, args: dict) -> Tuple[list, CommandResults]:
+    query_params = prepare_query_params(args)
+    events = client.get_logs(query_params)
+    results = CommandResults(
+        outputs_prefix='SlackEvents',
+        outputs_key_field='id',
+        outputs=events,
+        readable_output=tableToMarkdown('Slack Audit Logs', events, date_fields=['date_create']),
+    )
+    return events, results
+
+
+def fetch_events_command(client: Client, params: dict, last_run: dict) -> Tuple[list, dict]:
+    query_params = prepare_query_params(params)
+    query_params['oldest'] = last_run.get('oldest') or query_params.get('oldest')
+
+    if events := client.get_logs(query_params, last_run):
+        last_run.update({
+            'oldest': events[-1].get('date_create'),
+            'last_id': events[-1].get('id')
+        })
+    return events, last_run
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:  # pragma: no cover
+    command = demisto.command()
+    params = demisto.params()
+    args = demisto.args()
+
+    demisto.debug(f'Command being called is {command}')
+    try:
+        client = Client(
+            base_url=params.get('url'),
+            verify=not params.get('insecure'),
+            proxy=params.get('proxy'),
+            headers={
+                'Accept': 'application/json',
+                'Authorization': f'Bearer {params.pop("user_token", {}).get("password")}'
+            },
+        )
+
+        if command == 'test-module':
+            return_results(test_module_command(client))
+
+        else:
+            if command == 'slack-get-events':
+                events, results = get_events_command(client, args)
+                return_results(results)
+
+            else:  # command == 'fetch-events'
+                last_run = demisto.getLastRun()
+                events, last_run = fetch_events_command(client, params, last_run)
+                demisto.setLastRun(last_run)
+
+            if argToBoolean(params.get('should_push_events', 'true')):
+                send_events_to_xsiam(events, 'slack', 'slack')
```

Comment:
I talked with Anar about this, let's revert the vendor and product to be params again, will explain in private.